core/src/main/java/org/bitcoinj/core/Block.java

```diff
@@ -434,12 +434,14 @@ public String toString() {
      */
     @VisibleForTesting
     public void solve() {
+        if (Context.get().isRelaxProofOfWork())
```

Comment:
It sure would be nice to get rid of `Context.get()` here, if at all possible.

Reply:
Could we encode the fact that we are unit-testing and want relaxed PoW in the `Block` or `Difficulty` classes?

Reply:
I'd really like to get rid of it here too. I want to invest some time to see if we can make the affected unit tests "easier", meaning they use an easier difficulty.

I would rather not make "unit test mode" a property of Difficulty or Block.

Reply:
The main problem I have with this short-circuit is that we (implicitly) call solve() but at the same time we say "don't solve".

Reply:
Anyway, no reason to hold up the merge.